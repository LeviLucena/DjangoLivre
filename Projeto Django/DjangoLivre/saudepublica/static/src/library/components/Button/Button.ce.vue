<template>
  <button
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    v-bind="$attrs"
    class="br-button"
    :class="{
      primary: type === 'primary',
      secondary: type === 'secondary',
      large: density === 'large',
      middle: density === 'middle',
      small: density === 'small',
      xsmall: density === 'xsmall',
      loading,
      inverted,
      active,
      circle,
      block,
    }"
    aria-label=""
    :type="submit ? 'submit' : 'button'"
    :disabled="disabled"
  >
    <slot>
      <IconBase v-if="icon" :icon-name="icon" :class-name="getShowCircle"></IconBase>
      <span v-if="!circle">
        <slot>{{ label }}</slot>
      </span>
    </slot>
  </button>
</template>

<script>
import TooltipMixin from "../../mixins/TooltipMixin"
import IconBase from "../IconBase/IconBase.ce.vue"

const Button = {
  name: "brButton",
  components: {
    IconBase,
  },
  mixins: [TooltipMixin],
  props: {
    /**
     * **[OBRIGATÓRIO]** O rótulo é um elemento obrigatório. A **exceção** ocorre na versão tipo circular uma vez que não existe rótulo.
     */
    label: {
      type: String,
      default: null,
    },
    /**
     * **[OPCIONAL]** Ênfase do botão, com finalidade de proporcionar hierarquia visual a esses elementos e ações. Os botões mudam de aparência conforme a ênfase utilizada. A ênfase terciária é padrão. **[ primary | secondary ]**".
     */
    type: {
      type: String,
      validator: function (value) {
        return ["secondary", "primary", "tertiary"].includes(value)
      },
      default: null,
    },
    /**
     * **[Opcional]** Indica que o botão é do tipo *submit*.
     */
    submit: {
      type: Boolean,
      default: null,
    },
    /**
     * **[Opcional]** Tipo de botão muito usado dentro de outros componentes ou em situações onde a representação do ícone é suficiente para transmitir a mensagem ao usuário.
     */
    circle: {
      type: Boolean,
      default: false,
    },
    /** **[OPCIONAL]** O ajuste da densidade consiste em aumentar ou reduzir os espaçamentos internos do botão de forma a deixá-los mais compactos ou mais expandidos em seu formato. **[ large | middle | small ]**
     */
    density: {
      type: String,
      default: "middle",
      validator: function (value) {
        return ["large", "middle", "small", "xsmall"].includes(value)
      },
    },
    /** **[OPCIONAL]** Desabilita o botão */
    disabled: {
      type: Boolean,
      default: false,
    },
    /** **[OPCIONAL]** Apresenta o botão em comportamento de bloco. o botão “empurra” qualquer outro elemento existente para outro espaço disponível. */
    block: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Usado quando houver uma necessidade de enfatizar visualmente a ideia da ação. **Nos botões do tipo circular a utilização é obrigatória, visto que a mensagem da ação é transmitida apenas pelo ícone.**
     */
    icon: {
      type: [String],
      default: null,
    },
    /**
     * **[OPCIONAL]** Utilizado quando o background possui cores escuras.
     */
    inverted: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Utilizado para enfatizar o estado ativo do botão
     */
    active: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Aplica carregando
     */
    loading: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    getShowCircle() {
      return this.circle ? "" : "mr-1"
    },
  },
}

export default Button
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import "~@govbr-ds/core/src/components/button/_button";
@import "~@govbr-ds/core/src/components/loading/_loading";
@import "~@govbr-ds/core/src/components/notification/_notification";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
