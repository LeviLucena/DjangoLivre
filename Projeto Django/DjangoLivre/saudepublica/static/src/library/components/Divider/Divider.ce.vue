<template>
  <span class="br-divider" :class="classObject"></span>
</template>

<script>
const Divider = {
  name: "brDivider",
  props: {
    /**
     * Posiciona verticalmente o componente br-divider.
     *
     * Obs: A versão vertical só funciona caso o componente seja usado dentro de um elemento com [Flexbox](https://www.gov.br/ds/utilitarios/css/flexbox)
     */
    vertical: { type: Boolean, default: false },
    /**
     * **[OPCIONAL]** O ajuste consiste em aumentar ou reduzir o divider de forma a deixá-lo mais compacto ou mais expandido em seu formato. **[ small | medium| large ]**
     */
    size: {
      type: String,
      default: null,
      validator: function (value) {
        return ["small", "medium", "large"].includes(value)
      },
    },
    /**
     * **[OPCIONAL]** Deixa a linha do componente tracejada
     */
    dashed: { type: Boolean, default: false },
  },
  computed: {
    classObject() {
      return {
        sm: this.size === "small",
        md: this.size === "medium",
        lg: this.size === "large",
        vertical: this.vertical,
        dashed: this.dashed,
      }
    },
  },
}

export default Divider
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/components/divider/_mixins";
@import "~@govbr-ds/core/src/partial/scss/core-base.scss";

:host {
  .#{$prefix}divider,
  hr {
    @include divider-configs;
  }

  // Fix span vertical align
  .vertical {
    height: 100%;
  }
}
</style>
