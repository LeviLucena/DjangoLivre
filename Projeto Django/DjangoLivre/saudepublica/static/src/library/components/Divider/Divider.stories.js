import { kebabiseArgs } from '../../util/Utils.js'
import BrDivider from './Divider.ce.vue'

export default {
  title: 'Componentes/Divider',
  component: BrDivider,
  argTypes: {
    vertical: {
      // permite mostrar, mas desabilita a possibilidade de trocar o valor da propriedade [control: false]
      control: false,
      default: false,
    },
    size: {
      options: ['small', 'medium', 'large'],
      control: 'inline-radio',
      defaultValue: '',
    },
  },
}

const TemplateVertical = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `
  <div class="row">
    <div class="col">
      <div class="d-flex p-3">
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
        <br-divider v-bind="args" class="mx-3"></br-divider>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col">
      <div class="bg-secondary-07 text-secondary-01 p-3">
        <div class="d-flex">
          <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
          <br-divider v-bind="args" class="mx-3"></br-divider>
          <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
        </div>
      </div>
    </div>
  </div>
  `,
})

export const Vertical = TemplateVertical.bind({})
Vertical.args = {
  vertical: true,
}

Vertical.parameters = {
  controls: { exclude: [''] },
}

export const VerticalTracejado = TemplateVertical.bind({})
VerticalTracejado.args = {
  vertical: true,
  dashed: true,
}

VerticalTracejado.parameters = {
  controls: { exclude: [''] },
}
const TemplateHorizontal = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `

  <div class="row p-3">
    <div class="col">
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
      <br-divider v-bind="args" class="my-3"></br-divider>
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
    </div>
  </div>

  <div class="row">
  <div class="col">
    <div class="bg-secondary-07 text-secondary-01 p-3">
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
      <br-divider v-bind="args" class="my-3"></br-divider>
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
    </div>
  </div>
</div>

  `,
})

export const Horizontal = TemplateHorizontal.bind({})
Horizontal.args = {}
Horizontal.parameters = {
  controls: { exclude: ['vertical'] },
}

export const HorizontalTracejado = TemplateHorizontal.bind({})
HorizontalTracejado.args = {
  dashed: true,
}
HorizontalTracejado.parameters = {
  controls: { exclude: ['vertical'] },
}

export const Small = TemplateHorizontal.bind({})
Small.args = {
  size: 'small',
}
Small.parameters = {
  controls: { exclude: ['vertical'] },
}

export const Medium = TemplateHorizontal.bind({})
Medium.args = {
  size: 'medium',
}
Medium.parameters = {
  controls: { exclude: ['vertical'] },
}

export const Large = TemplateHorizontal.bind({})
Large.args = {
  size: 'large',
}
Large.parameters = {
  controls: { exclude: ['vertical'] },
}
