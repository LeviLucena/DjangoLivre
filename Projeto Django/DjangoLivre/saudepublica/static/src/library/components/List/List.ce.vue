<template>
  <div
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    class="br-list"
    :class="{ horizontal: horizontal, 'justify-content-between': spaceBetween }"
    role="list"
    :density="getDensity"
    :data-toggle="dataToggle"
    :data-unique="dataUnique"
  >
    <div v-if="title" class="header">
      <div class="title">
        {{ title }}
      </div>
    </div>
    <BrDivider v-if="title && !hideTitleDivider"></BrDivider>
    <slot></slot>
  </div>
</template>

<script>
import TooltipMixin from "../../mixins/TooltipMixin"
import BrDivider from "../Divider/Divider.ce.vue"

const List = {
  name: "brList",
  components: {
    BrDivider,
  },
  mixins: [TooltipMixin],
  inject: {
    parentDensity: {
      default: "small",
    },
  },
  provide() {
    return {
      parentList: this,
      parentDensity: this.getDensity,
    }
  },
  props: {
    /**
     * **[OPCIONAL]** Título da lista.
     * */
    title: {
      type: String,
      default: "",
    },
    /**
     * **[OPCIONAL]** Oculta o "divider" abaixo do título da lista.
     * */
    hideTitleDivider: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Indica se a lista será horizontal. por padrão, a lista é vertical.
     * */
    horizontal: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Indica que os itens da lista terão espaçamento uniforme entre eles.
     * */
    spaceBetween: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Ativa a lista expansiva, onde os \<br-item\> filhos abrem/fecham ao clicá-los.
     * */
    dataToggle: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Ativa a abertura de um \<br-item\> da lista por vez quando a mesma é data-toggle.
     * */
    dataUnique: {
      type: Boolean,
      default: false,
    },
    /**
     * TODO: documentação
     * */
    checkable: {
      type: Boolean,
      default: false,
    },
    /**
     * Niveis de densidade para o list, opções validas são "small", "medium" e "large"
     */
    density: {
      type: String,
      default: "large",
      validator: function (value) {
        return ["small", "medium", "large"].includes(value)
      },
    },
  },
  data() {
    return {
      items: [],
      getDensity: this.density,
    }
  },
  methods: {
    closeAllItems() {
      if (this.items && this.items.length) {
        this.items.forEach((item) => (item.isOpen = false))
      }
    },
    isDataToggle() {
      return this.dataToggle
    },
    isDataUnique() {
      return this.dataUnique
    },
    addItem(item) {
      return this.items.push(item)
    },
  },
}

export default List
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import "~@govbr-ds/core/src/components/list/_list";
@import "~@govbr-ds/core/src/components/divider/_divider";
</style>
