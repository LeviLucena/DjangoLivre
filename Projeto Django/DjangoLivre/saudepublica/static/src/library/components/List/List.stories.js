import { action } from '@storybook/addon-actions'
import { kebabiseArgs } from '../../util/Utils.js'
import BrList from './List.ce.vue'

export default {
  title: 'Componentes/List',
  component: BrList,
  argTypes: {
    title: {
      control: 'text',
      defaultValue: '',
    },
    hideTitleDivider: {
      control: 'boolean',
      defaultValue: false,
    },
    horizontal: {
      control: 'boolean',
      defaultValue: false,
    },
    dataToggle: {
      defaultValue: false,
    },
    dataUnique: {
      defaultValue: false,
    },
    checkable: {
      defaultValue: false,
    },
    default: {
      control: 'text',
      description:
        '**[OBRIGATÓRIO]** Conteúdo da lista, que deve ser passado por slot. Preferencialmente, devem ser do tipo **<br-item>**.',
      table: {
        type: {
          summary: 'NodeList of <br-item>',
        },
        defaultValue: {
          summary: '',
        },
      },
    },
    density: {
      control: 'inline-radio',
      options: ['large', 'medium', 'small'],
      defaultValue: 'small',
    },
  },
}

const TemplateDefault = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-list v-bind="args">
    <br-item title="Item 1" @toggle-open="handleOpen">
      <br-list>
        <br-item>Subitem 1.1</br-item>
        <br-item>Subitem 1.2</br-item>
        <br-item>Subitem 1.3</br-item>
      </br-list>
    </br-item>
    <br-item title="Item 2" @toggle-open="handleOpen">
      <br-list>
        <br-item>Detalhes do item 2</br-item>
      </br-list>
    </br-item>
    <br-item title="Item 3" @toggle-open="handleOpen">
      <br-list> Mais detalhes sobre o item 3 </br-list>
    </br-item>
    <br-item title="Item 4" @toggle-open="handleOpen">
      <br-list> Mais detalhes sobre o item 4 </br-list>
    </br-item>
  </br-list>`,
  methods: { handleOpen: action('toggle-open') },
})

const TemplateHorizontal = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-list v-bind="args">
    <br-item title="Item 1" @toggle-open="handleOpen">
      <br-list>
        <br-item>Subitem 1.1</br-item>
        <br-item>Subitem 1.2</br-item>
        <br-item>Subitem 1.3</br-item>
      </br-list>
    </br-item>
    <br-item title="Item 2" @toggle-open="handleOpen">
      <br-list>
        <br-item>Detalhes do item 2</br-item>
      </br-list>
    </br-item>
    <br-item title="Item 3" @toggle-open="handleOpen">
      <br-list> Mais detalhes sobre o item 3 </br-list>
    </br-item>
  </br-list>`,
  methods: { handleOpen: action('toggle-open') },
})

const TemplateComTooltip = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `
  <br-list v-bind="args">
    <br-item title="Item 1" tooltip-text="Lorem ipsum">
      <br-list>
        <br-item hover tooltip-text="Lorem ipsum">Subitem 1.1</br-item>
        <br-item hover>Subitem 1.2</br-item>
        <br-item hover>Subitem 1.3</br-item>
      </br-list>
    </br-item>
    <br-divider></br-divider>
    <br-item title="Item 2" tooltip-text="Lorem ipsum">
      <br-list>
        <br-item hover tooltip-text="Lorem ipsum">Subitem 2.1</br-item>
        <br-item hover>Subitem 2.2</br-item>
        <br-item hover>Subitem 2.3</br-item>
      </br-list>
    </br-item>
  </br-list>`,
})

const TemplateAgrupamentoPorSeparador = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-list v-bind="args">
    <br-item title="Item 1">
      <br-list>
        <br-item>Subitem 1.1</br-item>
        <br-item>Subitem 1.2</br-item>
        <br-item>Subitem 1.3</br-item>
      </br-list>
    </br-item>
    <br-divider></br-divider>
    <br-item title="Item 2">
      <br-list>
        <br-item>Detalhes do item 2</br-item>
      </br-list>
    </br-item>
    <br-divider></br-divider>
    <br-item title="Item 3">
      <br-list> Mais detalhes sobre o item 3 </br-list>
    </br-item>
    <br-divider></br-divider>
    <br-item title="Item 4">
      <br-list> Mais detalhes sobre o item 4 </br-list>
    </br-item>
  </br-list>`,
})

export const ListaVertical = TemplateDefault.bind({})
ListaVertical.args = {
  title: 'Lista vertical',
}

export const ListaHorizontal = TemplateHorizontal.bind({})
ListaHorizontal.args = {
  title: 'Lista horizontal',
  horizontal: true,
}
export const AgrupamentoPorExpansao = TemplateHorizontal.bind({})
AgrupamentoPorExpansao.args = {
  title: 'Agrupamento por expansão',
  horizontal: true,
  dataToggle: true,
}

export const AgrupamentoPorSeparador = TemplateAgrupamentoPorSeparador.bind({})
AgrupamentoPorSeparador.args = {
  title: 'Agrupamento por separador',
}

export const DataUnique = TemplateDefault.bind({})
DataUnique.args = {
  title: 'Isto é um <br-list data-unique dataToggle> que contém vários <br-item>',
  dataUnique: true,
  dataToggle: true,
}

export const DensityLarge = TemplateDefault.bind({})
DensityLarge.args = {
  title: 'Density Large',
  density: 'large',
}

export const DensityMedium = TemplateDefault.bind({})
DensityMedium.args = {
  title: 'Density Medium',
  density: 'medium',
}

export const DensitySmall = TemplateDefault.bind({})
DensitySmall.args = {
  title: 'Density Small',
}

export const ComTooltip = TemplateComTooltip.bind({})
ComTooltip.args = {
  title: 'Lista vertical',
  dataToggle: true,
}
