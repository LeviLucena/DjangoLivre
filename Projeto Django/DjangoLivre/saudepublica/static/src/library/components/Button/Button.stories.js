import { kebabiseArgs } from '../../util/Utils.js'
import Button from './Button.ce.vue'

export default {
  title: 'Componentes/Button',
  component: Button,
  parameters: {
    backgrounds: {
      default: 'light',
      values: [
        { name: 'light', value: '#fff' },
        { name: 'dark', value: '#071D41' },
      ],
    },
    actions: {
      handles: ['click'],
    },
  },
  argTypes: {
    label: {
      control: 'text',
    },
    type: {
      type: 'select',
      options: ['secondary', 'primary'],
    },
    density: {
      type: 'select',
      options: ['large', 'middle', 'small', 'xsmall'],
    },
    circle: {
      defaultValue: false,
    },
    disabled: {
      defaultValue: false,
    },
    block: {
      defaultValue: false,
    },
    icon: {
      defaultValue: '',
    },
    inverted: {
      defaultValue: false,
    },
    active: {
      defaultValue: false,
    },
    loading: {
      defaultValue: false,
    },
    tooltipText: {
      control: 'text',
    },
    tooltipPlace: {
      options: ['top', 'right', 'bottom', 'left'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipType: {
      options: ['info', 'success', 'warning', 'error'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipTimer: {
      control: 'number',
      if: { arg: 'tooltipText', truthy: true },
    },
    default: {
      control: 'text',
      description: 'Rótulo do botão, que pode ser passado por slot',
      table: {
        type: {
          summary: `string`,
        },
        defaultValue: {
          summary: '',
        },
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-button v-bind="args"></br-button>`,
})

const TemplateExternal = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-button v-bind="args">
  <br-icon-base icon-name="user"></br-icon-base>
  </br-button>`,
})

export const Primary = Template.bind({})
Primary.args = {
  label: 'Primary',
  type: 'primary',
}

export const Secondary = Template.bind({})
Secondary.args = {
  label: 'Secondary',
  type: 'secondary',
}

export const Tertiary = Template.bind({})
Tertiary.args = {
  label: 'Tertiary',
  inverted: false,
}

export const Circle = Template.bind({})
Circle.parameters = { controls: { exclude: ['label', 'block'] } }
Circle.args = {
  type: 'primary',
  circle: true,
  icon: 'trash',
  inverted: false,
}

export const Loading = Template.bind({})
Loading.parameters = {
  controls: { exclude: ['label', 'block', 'disabled', 'icon', 'circle'] },
}
Loading.args = {
  type: 'secondary',
  loading: true,
}

export const Inverted = Template.bind({})
Inverted.parameters = { backgrounds: { default: 'dark' } }
Inverted.args = {
  label: 'Inverted',
  type: 'secondary',
  inverted: true,
}

export const IconAndLabel = Template.bind({})
IconAndLabel.args = {
  label: 'Text',
  type: 'secondary',
  icon: 'city',
  inverted: false,
}

export const IconAndCircle = Template.bind({})
IconAndCircle.parameters = { controls: { exclude: ['label', 'block'] } }
IconAndCircle.args = {
  circle: true,
  type: 'secondary',
  icon: 'city',
  inverted: false,
}

export const IconAndCircleExternal = TemplateExternal.bind({})
IconAndCircleExternal.parameters = { controls: { exclude: ['label', 'block'] } }
IconAndCircleExternal.args = {
  circle: true,
  type: 'secondary',
  icon: 'city',
  inverted: false,
}

export const ComTooltip = Template.bind({})
ComTooltip.args = {
  label: 'Botão com tooltip',
  type: 'primary',
  tooltipText: 'Lorem ipsum dolor sit amet.',
  tooltipType: 'warning',
}
