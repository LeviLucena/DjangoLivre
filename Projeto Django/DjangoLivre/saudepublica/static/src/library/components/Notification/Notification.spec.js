import { shallowMount } from '@vue/test-utils'
import Button from './../Button/Button.ce.vue'
import Divider from './../Divider/Divider.ce.vue'
import List from './../List/List.ce.vue'
import Tab from './../Tab/Tab.ce.vue'
import TabItem from './../Tab/TabItem.ce.vue'
import BrNotification from './Notification.ce.vue'

describe('br-notification', () => {
  const showNotification = true
  const titleExample = 'title example'
  const subtitleExample = 'subtitle example'
  const notificationsItens = `[
    {
      tab: { icon: 'bell' },
      itens: [
        {icon: 'heartbeat', text: 'Link disable', state: 'disabled'},
        {icon: 'heartbeat', text: 'Link de acesso'},
        {icon: 'heartbeat', text: 'Link Google', href: 'https://www.google.com'}
      ]
    },
    {
      tab: { icon: 'envelope' },
      itens: [
        {icon: 'heartbeat', text: 'Link disable', state: 'disabled'},
        {icon: 'heartbeat', text: 'Link de acesso'},
        {icon: 'heartbeat', text: 'Link Google', href: 'https://www.google.com'}
      ]
    }
  ]`

  const globalConfig = {
    stubs: {
      'br-button': Button,
      'br-divider': Divider,
      'br-tab': Tab,
      'br-tab-item': TabItem,
      'br-list': List,
    },
  }

  test('check if renders class br-notification', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    expect(wrapper.find('.br-notification').exists()).toBe(true)
  })

  test('check if renders class notification-header ', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    expect(wrapper.find('.notification-header').exists()).toBe(true)
  })

  test('check if renders class notification-body ', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    expect(wrapper.find('.notification-body').exists()).toBe(true)
  })

  test('check if prop title its visible.', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    expect(wrapper.text()).toMatch(titleExample)
  })

  test('check if prop subtitle its visible.', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    expect(wrapper.text()).toMatch(subtitleExample)
  })

  test('check if notifications are visible', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })

    expect(wrapper.findAll('.br-item')).toHaveLength(6)
  })

  test('check if close button its push', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    const closeNotification = wrapper.find('.col-2')
    closeNotification.trigger('click')
    wrapper.vm.close()
  })

  test('check notification without header', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        disableCloseButton: true,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    expect(wrapper.find('.notification-header').exists()).toBe(false)
  })

  test('disableCloseButton hides close button', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: { showNotification, disableCloseButton: true, notifications: notificationsItens },
      global: globalConfig,
    })
    expect(wrapper.find('.text-right').exists()).toBe(false)
  })

  test('matches snapshot', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('close button is clicked', () => {
    const wrapper = shallowMount(BrNotification, {
      propsData: {
        showNotification: showNotification,
        title: titleExample,
        subtitle: subtitleExample,
        notifications: notificationsItens,
      },
      global: globalConfig,
    })
    const closeBtn = wrapper.find('#close')
    closeBtn.trigger('click')
    expect(wrapper.emitted('close')).toBeTruthy()
  })
})
