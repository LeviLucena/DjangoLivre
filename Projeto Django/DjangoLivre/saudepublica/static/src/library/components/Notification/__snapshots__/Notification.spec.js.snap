// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`br-notification matches snapshot 1`] = `
<div
  class="br-notification"
>
  <div
    class="notification-header"
  >
    <div
      class="row"
    >
      <div
        class="col-10"
      >
        
        <span
          class="text-bold"
        >
          title example
        </span>
        <br />
        
        
        <small>
          <a>
            subtitle example
          </a>
        </small>
        <br />
        
        
        
      </div>
      <div
        class="col-2"
      >
        <div
          class="text-right"
        >
          <button
            aria-label="Fechar"
            class="br-button middle circle"
            id="close"
            type="button"
          >
            
            <icon-base-stub
              class=""
              familyname="fas"
              iconname="times"
            />
            <!--v-if-->
            
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="notification-body"
  >
    <div
      class="br-tab small"
      data-counter="false"
    >
      <nav
        aria-label="Abas de Navegação Simples"
        class="tab-nav"
      >
        <ul>
          
          
        </ul>
      </nav>
      <div
        class="tab-content"
      >
        
        
        <div
          class="tab-panel"
          id=""
          style="display: none;"
        >
          
          <div
            class="br-list"
            data-toggle="false"
            data-unique="false"
            density="large"
            role="list"
          >
            <!--v-if-->
            <!--v-if-->
            
            
            <div>
              <button
                class="br-item disabled"
                type="button"
              >
                <!--v-if-->
                <!--v-if-->
                <!--v-if-->
                <icon-base-stub
                  class=""
                  familyname="fas"
                  iconname="heartbeat"
                />
                 Link disable
              </button>
              <span
                class="br-divider"
              />
            </div>
            <div>
              <button
                class="br-item"
                type="button"
              >
                <!--v-if-->
                <!--v-if-->
                <!--v-if-->
                <icon-base-stub
                  class=""
                  familyname="fas"
                  iconname="heartbeat"
                />
                 Link de acesso
              </button>
              <span
                class="br-divider"
              />
            </div>
            <div>
              <a
                href="https://www.google.com"
              >
                <button
                  class="br-item"
                  type="button"
                >
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                  <icon-base-stub
                    class=""
                    familyname="fas"
                    iconname="heartbeat"
                  />
                   Link Google
                </button>
              </a>
              <span
                class="br-divider"
              />
            </div>
            
            
          </div>
          
        </div>
        <div
          class="tab-panel"
          id=""
          style="display: none;"
        >
          
          <div
            class="br-list"
            data-toggle="false"
            data-unique="false"
            density="large"
            role="list"
          >
            <!--v-if-->
            <!--v-if-->
            
            
            <div>
              <button
                class="br-item disabled"
                type="button"
              >
                <!--v-if-->
                <!--v-if-->
                <!--v-if-->
                <icon-base-stub
                  class=""
                  familyname="fas"
                  iconname="heartbeat"
                />
                 Link disable
              </button>
              <span
                class="br-divider"
              />
            </div>
            <div>
              <button
                class="br-item"
                type="button"
              >
                <!--v-if-->
                <!--v-if-->
                <!--v-if-->
                <icon-base-stub
                  class=""
                  familyname="fas"
                  iconname="heartbeat"
                />
                 Link de acesso
              </button>
              <span
                class="br-divider"
              />
            </div>
            <div>
              <a
                href="https://www.google.com"
              >
                <button
                  class="br-item"
                  type="button"
                >
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                  <icon-base-stub
                    class=""
                    familyname="fas"
                    iconname="heartbeat"
                  />
                   Link Google
                </button>
              </a>
              <span
                class="br-divider"
              />
            </div>
            
            
          </div>
          
        </div>
        
        
      </div>
    </div>
  </div>
</div>
`;
